### ISSUE TITLE
The remote server returned an error: (400) Bad Request. Bad Request {"error":{"code":"Unknown","message":"The specified file is not an extension file. (Parameter 'stream') 

### ISSUE THREAD

**Issue by PhilippeCabel** [microsoft/navcontainerhelper#3659]
Hello
Since last Friday (30/08/2024), we've been experiencing the following problem when trying to deploy applications in a Saas environment from Git.
`"The remote server returned an error: (400) Bad Request. Bad Request {"error":{"code":"Unknown","message":"The specified file is not an extension file. (Parameter 'stream')"
`
![image](https://github.com/user-attachments/assets/11f1fda6-35ea-40b5-a86f-0382dac12c52)

After some research, the problem seems to arise when the Publish-PerTenantExtensionApps.ps1 script executes the following Invoke_RestMethod
```
 Invoke-RestMethod `
        -Method Post `
        -Uri "$automationApiUrl/companies($companyId)/extensionUpload($($extensionUpload.systemId))/Microsoft.NAV.upload" `
        -Headers ((GetAuthHeaders) + $ifMatchHeader) | Out-Null
```
![image](https://github.com/user-attachments/assets/3491c5d2-a308-4f59-804c-dec97e64eecd)

Building applications with the runpipeline works well, and if we upload the generated applications using the ‘upload extensions’ button in extension management, that works too. Sometimes 1 or 2 extensions go through.

Thanks for your help

**Comment by freddydk** [comment]
If this was a problem with Publish-PerTenantExtensionApps - I would assume that a ton of people would be complaining.
When you say that you can manually upload the app - have you tried to run Publish-PerTenantExtensionApps (manually) and upload the app?

**Comment by PhilippeCabel** [comment]
Hello Freddyk
I can imagine that :) for the complaining.
Yes I have already tested a manual upload with a separate powershell with the same result.
Is it possible this issue could be linked to the powershell version?
Thanks

**Comment by PhilippeCabel** [comment]
Hello @esbenk 
I did what you explained without any problem.

**Comment by PhilippeCabel** [comment]
Hello @freddydk and @esbenk 
I have try with the version 5.0 and 7.0 of powershell with the same issue => bad request.
I have try also with a dummy app (generated by AL:GO!) with same issue => bad request.
if you have an idea, I'd love to hear from you :)
By the way, thanks for your time !

**Comment by freddydk** [comment]
What version of BcContainerHelper are you using?
What runners are you using (self-hosted or github hosted)?
Is there a anti-virus product on the runner (if self-hosted)?

**Comment by mazhelez** [comment]
@PhilippeCabel Can you give me read access to the repository in question? From the related incident, I see the failure is from a custom pipeline.

**Comment by freddydk** [comment]
Have you tried to download the app and use the Publish-PerTenantExtensionApps on you local computer with the same result?

This would tell you whether the problem is in this function (which I do not think) or the problem is in your code somewhere.

If this also fails, then you could share the .app file with us in an email together with the script used to repro the problem

**Comment by PhilippeCabel** [comment]
Hello @freddydk 
Have you tried to download the app and use the Publish-PerTenantExtensionApps on you local computer with the same result? => Yes with same result.
I also try to upload a hello world app generated by "AL:GO!" with the same result.
But what's strange is that sometimes I've managed to download certain applications but not all of them. And if I try in different environments, the problem can occur on different applications.

**Comment by PhilippeCabel** [comment]
> @PhilippeCabel Can you give me read access to the repository in question? From the related incident, I see the failure is from a custom pipeline.

**Comment by MadaPR** [comment]
hi @mazhelez you have now read access to our repo. I guess you have received the invite.

**Comment by mazhelez** [comment]
@PhilippeCabel I'd have to wait with accepting the invitation for now.

Can you please go on with testing as per Freddy's reccommendation:
>Have you tried to download the app and use the Publish-PerTenantExtensionApps on you local computer with the same result?

>This would tell you whether the problem is in this function (which I do not think) or the problem is in your code somewhere.

If you feel like the best way is to share anything, can you please create a public repository with a repro of the issue? Make sure to not include any IP in that repository.

**Comment by MadaPR** [comment]
hi @mazhelez @freddydk , 

do you need more elements from our side? it is not clear to me if you are waiting from some action from our part or if you have enough to pursue investigations on your side?
we have opened this in relation with a MS support case ( https://serviceshub.microsoft.com/support/case/2409050050000680 ) and we are notified every day with the message saying that investigation is on going and that we'll have updates the next day.

we are stuck for more than a week now and our project is heavily impacted by this issue. 
how can we make progress on it ?

**Comment by PhilippeCabel** [comment]
> @PhilippeCabel I'd have to wait with accepting the invitation for now.
> 
> Can you please go on with testing as per Freddy's reccommendation:
> 
> > Have you tried to download the app and use the Publish-PerTenantExtensionApps on you local computer with the same result?
> 
> > This would tell you whether the problem is in this function (which I do not think) or the problem is in your code somewhere.
> 
> If you feel like the best way is to share anything, can you please create a public repository with a repro of the issue? Make sure to not include any IP in that repository.

Like I said before, I have already test that with the same result

**Comment by MadaPR** [comment]
hi @freddydk i think the best course of action would be to have a meeting so that we can show you the behaviour and asses together the situation. 
could we arrange a meeting tomorrow 13th of september ?

**Comment by freddydk** [comment]
I am attending Days Of Knowledge Americas 2024 starting tomorrow morning and cannot attend any meeting.

Also before we have a meeting, I would need to see the exact code used to repro the problem on your side (like I have showed the code, which apparently does not repro the problem).

The code needs to be extracted from all other code - basically like the two lines I shared - and the output given from that code and you can try with the app I created here: [Default Publisher_ALProject15_1.0.0.0.zip](https://github.com/user-attachments/files/16982159/Default.Publisher_ALProject15_1.0.0.0.zip)

**Comment by mazhelez** [comment]
@PhilippeCabel @MadaPR from the screenshot in the incident, I see it's your custom workflow failing (CICD_DeployPrBuild.yaml).  Do you also repro the issue with the CICD.yaml (the original one from AL-Go for GitHub)?

**Comment by mazhelez** [comment]
Also, if possible, can you share the whole log of the pipeline run? There might be something wrong with the way the app is built.

**Comment by PhilippeCabel** [comment]
Hello
I've found the root cause of our problem. In our applications, we have a feature that allows objects to be activated/deactivated using the application area. One of our developers has introduced a codeunit with a subscriber to the ‘OnAfterLogin’ event in the ‘System Initialization’ codeunit. And in this subscriber function, we refresh these specific application areas.
This explains why it works well when we use the download function from extension management because we're already logged in.
We can now close this topic and thank you for your help.

**Comment by yurim108** [comment]
@PhilippeCabel what was the solution? 

I am facing a similar problem. I am getting exactly the same error message on one of the customer's sandboxes (Test). 
Installing the same app on another sandbox (Dev) and in production environment using API works.
Installing the same app manually on the Test sandbox works.

I have tried with two different apps. One app has an OnAfterLogin subscriber since years and the second app has no such subscriber. Installation of both apps fail in the same way. It worked before the update from version 27.2.42879.42994 to 27.3.44313.44463.  

At the moment both Dev and Test environments have BC version 27.5.46862.46929.

I have also tried to delete the record in the "API Extension Upload" table and install the app using API again: same result.
I have checked the content of the blob field in the "API Extension Upload" table: it has exactly the same length as the app file. 
The problem occurs obviously when the "Upload" function of the "APIV2 - Aut. Extension Upload" page is called, which in the end calls:
DotNetALNavAppOperationInvoker.UploadPackage(PackageInStream, DotNetALPackageDeploymentSchedule, Format(lcid), DotNetALNavAppSyncMode);

The debugger didn't catch anything.

**Comment by ChrisePac** [comment]
@yurim108  have you had any feedback?
I ran now in exactly the same issue.
1 out of my 18 extensions have the same error message now. I can't deploy to any of my sandboxes.

As we are using AlOps I switched recently from their V1 compiler to the V3 compiler.
But I can repro the beaviour with VSCode/Cursor and the new ALC compile command.

**Comment by ChrisePac** [comment]
@mazhelez  & @esbenk I think I found the issue in our case at least.
Maybe thats intersting for you @esbenk , but I assume you know allready.

I had a vendor name framed in round brackets '(xxx)' in my extension name.
When I now compare 17.0.2273547 vs 18.0.2498801 the result is a difference in TextData and Translations output folder.
17.x creates the files there with '(xxx)', but 18.x now creates them with '%28xxx%29' in the file names.

I removed the whole (xxx) from the app.json now and since then the extension can be deployed.

Hope that helps.

**Comment by yurim108** [comment]
@mazhelez @esbenk @ChrisePac I have found the reason of the problem in my case. 
Microsoft confirmed it to be a bug and fixed it in one of the latest BC 28.2 build. Microsoft refused to port the fix to older BC versions. 

<img width="1285" height="454" alt="Image" src="https://github.com/user-attachments/assets/4de95501-7136-47fe-865d-e55763da8da0" />
